llvm/lib/Target/AMDGPU/SIInstructions.td

```diff
@@ -3553,6 +3553,23 @@ def : AMDGPUPat <
   (V_BFE_U32_e64 $src, (i32 0), $width)
 >;
 
+def uint5Bits : PatLeaf<(i32 VGPR_32:$width), [{
+  return CurDAG->computeKnownBits(SDValue(N, 0)).countMaxTrailingOnes() <= 5;
```

Comment:
```suggestion
  return CurDAG->computeKnownBits(SDValue(N, 0)).countMaxActiveBits() <= 5;
```
Please also add a test that shows why countMaxTrailingOnes was wrong.

Reply:
Thanks, Done in https://github.com/llvm/llvm-project/pull/115543